rust/core/src/lib.rs

```diff
@@ -64,32 +76,29 @@ pub type Mint = OrderedHashMap<PolicyId, OrderedHashMap<AssetName, Int64>>;
 
 pub type Multiasset = OrderedHashMap<PolicyId, OrderedHashMap<AssetName, u64>>;
 
+// TODO for regen: why was this in babbage/crypto.cddl? it's not in babbage.cddl.
 pub type Natural = Vec<u8>;
 
+// we might want dedicated types for this anyway?
 pub type PlutusV1Script = Vec<u8>;
 
 pub type PlutusV2Script = Vec<u8>;
 
-pub type PolicyId = Hash28;
+pub type PolicyId = ScriptHash;
 
 pub type PolicyIds = Vec<PolicyId>;
 
-pub type PoolKeyhash = Hash28;
-
-pub type PoolMetadataHash = Hash32;
-
 pub type Port = u16;
 
-pub type ProposedProtocolParameterUpdates = OrderedHashMap<Genesishash, ProtocolParamUpdate>;
+pub type ProposedProtocolParameterUpdates = OrderedHashMap<GenesisHash, ProtocolParamUpdate>;
 
 pub type RewardAccounts = Vec<RewardAccount>;
 
-pub type ScriptDataHash = Hash32;
-
+// TODO: this should NOT be generating an alias! (it's both tagged and a .cbor script)
+// we should investigate cddl-codegen and open an issue there if it's still happening
+// (top-level .cbor was recently being worked on and this is tagged but even then it should NOT be bytes here)
 pub type ScriptRef = Vec<u8>;
```

Comment:
Issue: [cddl-codegen#117](https://github.com/dcSpark/cddl-codegen/issues/117)

Reply:
should be resolved already